diff --git a/lib/hostip.c b/lib/hostip.c
index d721403b7237c2..8ab183af1b39e8 100644
--- a/lib/hostip.c
+++ b/lib/hostip.c
@@ -682,13 +682,7 @@ enum resolve_t Curl_resolv(struct Curl_easy *data,
   CURLcode result;
   enum resolve_t rc = CURLRESOLV_ERROR; /* default to failure */
   struct connectdata *conn = data->conn;
-  /* We should intentionally error and not resolve .onion TLDs */
-  size_t hostname_len = strlen(hostname);
-  if(hostname_len >= 7 &&
-     (curl_strequal(&hostname[hostname_len - 6], ".onion") ||
-      curl_strequal(&hostname[hostname_len - 7], ".onion."))) {
-    failf(data, "Not resolving .onion address (RFC 7686)");
-    return CURLRESOLV_ERROR;
-  }
+
   *entry = NULL;
 #ifndef CURL_DISABLE_DOH
